// eslint/lib/rules/index.js
/**
 * @fileoverview Collects the built-in rules into a map structure so that they can be imported all at once and without
 * using the file-system directly.
 * @author Peter (Somogyvari) Metz
 */

"use strict";

/* eslint sort-keys: ["error", "asc"] */

const {LazyLoadingRuleMap} = require("eslint/lib/rules/utils/lazy-loading-rule-map");

/** @type {Map<string, import("eslint/lib/shared/types").Rule>} */
module.exports = new LazyLoadingRuleMap(Object.entries({
    "accessor-pairs": () => require("eslint/lib/rules/accessor-pairs"),
    "array-bracket-newline": () => require("eslint/lib/rules/array-bracket-newline"),
    "array-bracket-spacing": () => require("eslint/lib/rules/array-bracket-spacing"),
    "array-callback-return": () => require("eslint/lib/rules/array-callback-return"),
    "array-element-newline": () => require("eslint/lib/rules/array-element-newline"),
    "arrow-body-style": () => require("eslint/lib/rules/arrow-body-style"),
    "arrow-parens": () => require("eslint/lib/rules/arrow-parens"),
    "arrow-spacing": () => require("eslint/lib/rules/arrow-spacing"),
    "block-scoped-var": () => require("eslint/lib/rules/block-scoped-var"),
    "block-spacing": () => require("eslint/lib/rules/block-spacing"),
    "brace-style": () => require("eslint/lib/rules/brace-style"),
    "callback-return": () => require("eslint/lib/rules/callback-return"),
    camelcase: () => require("eslint/lib/rules/camelcase"),
    "capitalized-comments": () => require("eslint/lib/rules/capitalized-comments"),
    "class-methods-use-this": () => require("eslint/lib/rules/class-methods-use-this"),
    "comma-dangle": () => require("eslint/lib/rules/comma-dangle"),
    "comma-spacing": () => require("eslint/lib/rules/comma-spacing"),
    "comma-style": () => require("eslint/lib/rules/comma-style"),
    complexity: () => require("eslint/lib/rules/complexity"),
    "computed-property-spacing": () => require("eslint/lib/rules/computed-property-spacing"),
    "consistent-return": () => require("eslint/lib/rules/consistent-return"),
    "consistent-this": () => require("eslint/lib/rules/consistent-this"),
    "constructor-super": () => require("eslint/lib/rules/constructor-super"),
    curly: () => require("eslint/lib/rules/curly"),
    "default-case": () => require("eslint/lib/rules/default-case"),
    "default-param-last": () => require("eslint/lib/rules/default-param-last"),
    "dot-location": () => require("eslint/lib/rules/dot-location"),
    "dot-notation": () => require("eslint/lib/rules/dot-notation"),
    "eol-last": () => require("eslint/lib/rules/eol-last"),
    eqeqeq: () => require("eslint/lib/rules/eqeqeq"),
    "for-direction": () => require("eslint/lib/rules/for-direction"),
    "func-call-spacing": () => require("eslint/lib/rules/func-call-spacing"),
    "func-name-matching": () => require("eslint/lib/rules/func-name-matching"),
    "func-names": () => require("eslint/lib/rules/func-names"),
    "func-style": () => require("eslint/lib/rules/func-style"),
    "function-call-argument-newline": () => require("eslint/lib/rules/function-call-argument-newline"),
    "function-paren-newline": () => require("eslint/lib/rules/function-paren-newline"),
    "generator-star-spacing": () => require("eslint/lib/rules/generator-star-spacing"),
    "getter-return": () => require("eslint/lib/rules/getter-return"),
    "global-require": () => require("eslint/lib/rules/global-require"),
    "grouped-accessor-pairs": () => require("eslint/lib/rules/grouped-accessor-pairs"),
    "guard-for-in": () => require("eslint/lib/rules/guard-for-in"),
    "handle-callback-err": () => require("eslint/lib/rules/handle-callback-err"),
    "id-blacklist": () => require("eslint/lib/rules/id-blacklist"),
    "id-length": () => require("eslint/lib/rules/id-length"),
    "id-match": () => require("eslint/lib/rules/id-match"),
    "implicit-arrow-linebreak": () => require("eslint/lib/rules/implicit-arrow-linebreak"),
    indent: () => require("eslint/lib/rules/indent"),
    "indent-legacy": () => require("eslint/lib/rules/indent-legacy"),
    "init-declarations": () => require("eslint/lib/rules/init-declarations"),
    "jsx-quotes": () => require("eslint/lib/rules/jsx-quotes"),
    "key-spacing": () => require("eslint/lib/rules/key-spacing"),
    "keyword-spacing": () => require("eslint/lib/rules/keyword-spacing"),
    "line-comment-position": () => require("eslint/lib/rules/line-comment-position"),
    "linebreak-style": () => require("eslint/lib/rules/linebreak-style"),
    "lines-around-comment": () => require("eslint/lib/rules/lines-around-comment"),
    "lines-around-directive": () => require("eslint/lib/rules/lines-around-directive"),
    "lines-between-class-members": () => require("eslint/lib/rules/lines-between-class-members"),
    "max-classes-per-file": () => require("eslint/lib/rules/max-classes-per-file"),
    "max-depth": () => require("eslint/lib/rules/max-depth"),
    "max-len": () => require("eslint/lib/rules/max-len"),
    "max-lines": () => require("eslint/lib/rules/max-lines"),
    "max-lines-per-function": () => require("eslint/lib/rules/max-lines-per-function"),
    "max-nested-callbacks": () => require("eslint/lib/rules/max-nested-callbacks"),
    "max-params": () => require("eslint/lib/rules/max-params"),
    "max-statements": () => require("eslint/lib/rules/max-statements"),
    "max-statements-per-line": () => require("eslint/lib/rules/max-statements-per-line"),
    "multiline-comment-style": () => require("eslint/lib/rules/multiline-comment-style"),
    "multiline-ternary": () => require("eslint/lib/rules/multiline-ternary"),
    "new-cap": () => require("eslint/lib/rules/new-cap"),
    "new-parens": () => require("eslint/lib/rules/new-parens"),
    "newline-after-var": () => require("eslint/lib/rules/newline-after-var"),
    "newline-before-return": () => require("eslint/lib/rules/newline-before-return"),
    "newline-per-chained-call": () => require("eslint/lib/rules/newline-per-chained-call"),
    "no-alert": () => require("eslint/lib/rules/no-alert"),
    "no-array-constructor": () => require("eslint/lib/rules/no-array-constructor"),
    "no-async-promise-executor": () => require("eslint/lib/rules/no-async-promise-executor"),
    "no-await-in-loop": () => require("eslint/lib/rules/no-await-in-loop"),
    "no-bitwise": () => require("eslint/lib/rules/no-bitwise"),
    "no-buffer-constructor": () => require("eslint/lib/rules/no-buffer-constructor"),
    "no-caller": () => require("eslint/lib/rules/no-caller"),
    "no-case-declarations": () => require("eslint/lib/rules/no-case-declarations"),
    "no-catch-shadow": () => require("eslint/lib/rules/no-catch-shadow"),
    "no-class-assign": () => require("eslint/lib/rules/no-class-assign"),
    "no-compare-neg-zero": () => require("eslint/lib/rules/no-compare-neg-zero"),
    "no-cond-assign": () => require("eslint/lib/rules/no-cond-assign"),
    "no-confusing-arrow": () => require("eslint/lib/rules/no-confusing-arrow"),
    "no-console": () => require("eslint/lib/rules/no-console"),
    "no-const-assign": () => require("eslint/lib/rules/no-const-assign"),
    "no-constant-condition": () => require("eslint/lib/rules/no-constant-condition"),
    "no-constructor-return": () => require("eslint/lib/rules/no-constructor-return"),
    "no-continue": () => require("eslint/lib/rules/no-continue"),
    "no-control-regex": () => require("eslint/lib/rules/no-control-regex"),
    "no-debugger": () => require("eslint/lib/rules/no-debugger"),
    "no-delete-var": () => require("eslint/lib/rules/no-delete-var"),
    "no-div-regex": () => require("eslint/lib/rules/no-div-regex"),
    "no-dupe-args": () => require("eslint/lib/rules/no-dupe-args"),
    "no-dupe-class-members": () => require("eslint/lib/rules/no-dupe-class-members"),
    "no-dupe-else-if": () => require("eslint/lib/rules/no-dupe-else-if"),
    "no-dupe-keys": () => require("eslint/lib/rules/no-dupe-keys"),
    "no-duplicate-case": () => require("eslint/lib/rules/no-duplicate-case"),
    "no-duplicate-imports": () => require("eslint/lib/rules/no-duplicate-imports"),
    "no-else-return": () => require("eslint/lib/rules/no-else-return"),
    "no-empty": () => require("eslint/lib/rules/no-empty"),
    "no-empty-character-class": () => require("eslint/lib/rules/no-empty-character-class"),
    "no-empty-function": () => require("eslint/lib/rules/no-empty-function"),
    "no-empty-pattern": () => require("eslint/lib/rules/no-empty-pattern"),
    "no-eq-null": () => require("eslint/lib/rules/no-eq-null"),
    "no-eval": () => require("eslint/lib/rules/no-eval"),
    "no-ex-assign": () => require("eslint/lib/rules/no-ex-assign"),
    "no-extend-native": () => require("eslint/lib/rules/no-extend-native"),
    "no-extra-bind": () => require("eslint/lib/rules/no-extra-bind"),
    "no-extra-boolean-cast": () => require("eslint/lib/rules/no-extra-boolean-cast"),
    "no-extra-label": () => require("eslint/lib/rules/no-extra-label"),
    "no-extra-parens": () => require("eslint/lib/rules/no-extra-parens"),
    "no-extra-semi": () => require("eslint/lib/rules/no-extra-semi"),
    "no-fallthrough": () => require("eslint/lib/rules/no-fallthrough"),
    "no-floating-decimal": () => require("eslint/lib/rules/no-floating-decimal"),
    "no-func-assign": () => require("eslint/lib/rules/no-func-assign"),
    "no-global-assign": () => require("eslint/lib/rules/no-global-assign"),
    "no-implicit-coercion": () => require("eslint/lib/rules/no-implicit-coercion"),
    "no-implicit-globals": () => require("eslint/lib/rules/no-implicit-globals"),
    "no-implied-eval": () => require("eslint/lib/rules/no-implied-eval"),
    "no-import-assign": () => require("eslint/lib/rules/no-import-assign"),
    "no-inline-comments": () => require("eslint/lib/rules/no-inline-comments"),
    "no-inner-declarations": () => require("eslint/lib/rules/no-inner-declarations"),
    "no-invalid-regexp": () => require("eslint/lib/rules/no-invalid-regexp"),
    "no-invalid-this": () => require("eslint/lib/rules/no-invalid-this"),
    "no-irregular-whitespace": () => require("eslint/lib/rules/no-irregular-whitespace"),
    "no-iterator": () => require("eslint/lib/rules/no-iterator"),
    "no-label-var": () => require("eslint/lib/rules/no-label-var"),
    "no-labels": () => require("eslint/lib/rules/no-labels"),
    "no-lone-blocks": () => require("eslint/lib/rules/no-lone-blocks"),
    "no-lonely-if": () => require("eslint/lib/rules/no-lonely-if"),
    "no-loop-func": () => require("eslint/lib/rules/no-loop-func"),
    "no-magic-numbers": () => require("eslint/lib/rules/no-magic-numbers"),
    "no-misleading-character-class": () => require("eslint/lib/rules/no-misleading-character-class"),
    "no-mixed-operators": () => require("eslint/lib/rules/no-mixed-operators"),
    "no-mixed-requires": () => require("eslint/lib/rules/no-mixed-requires"),
    "no-mixed-spaces-and-tabs": () => require("eslint/lib/rules/no-mixed-spaces-and-tabs"),
    "no-multi-assign": () => require("eslint/lib/rules/no-multi-assign"),
    "no-multi-spaces": () => require("eslint/lib/rules/no-multi-spaces"),
    "no-multi-str": () => require("eslint/lib/rules/no-multi-str"),
    "no-multiple-empty-lines": () => require("eslint/lib/rules/no-multiple-empty-lines"),
    "no-native-reassign": () => require("eslint/lib/rules/no-native-reassign"),
    "no-negated-condition": () => require("eslint/lib/rules/no-negated-condition"),
    "no-negated-in-lhs": () => require("eslint/lib/rules/no-negated-in-lhs"),
    "no-nested-ternary": () => require("eslint/lib/rules/no-nested-ternary"),
    "no-new": () => require("eslint/lib/rules/no-new"),
    "no-new-func": () => require("eslint/lib/rules/no-new-func"),
    "no-new-object": () => require("eslint/lib/rules/no-new-object"),
    "no-new-require": () => require("eslint/lib/rules/no-new-require"),
    "no-new-symbol": () => require("eslint/lib/rules/no-new-symbol"),
    "no-new-wrappers": () => require("eslint/lib/rules/no-new-wrappers"),
    "no-obj-calls": () => require("eslint/lib/rules/no-obj-calls"),
    "no-octal": () => require("eslint/lib/rules/no-octal"),
    "no-octal-escape": () => require("eslint/lib/rules/no-octal-escape"),
    "no-param-reassign": () => require("eslint/lib/rules/no-param-reassign"),
    "no-path-concat": () => require("eslint/lib/rules/no-path-concat"),
    "no-plusplus": () => require("eslint/lib/rules/no-plusplus"),
    "no-process-env": () => require("eslint/lib/rules/no-process-env"),
    "no-process-exit": () => require("eslint/lib/rules/no-process-exit"),
    "no-proto": () => require("eslint/lib/rules/no-proto"),
    "no-prototype-builtins": () => require("eslint/lib/rules/no-prototype-builtins"),
    "no-redeclare": () => require("eslint/lib/rules/no-redeclare"),
    "no-regex-spaces": () => require("eslint/lib/rules/no-regex-spaces"),
    "no-restricted-globals": () => require("eslint/lib/rules/no-restricted-globals"),
    "no-restricted-imports": () => require("eslint/lib/rules/no-restricted-imports"),
    "no-restricted-modules": () => require("eslint/lib/rules/no-restricted-modules"),
    "no-restricted-properties": () => require("eslint/lib/rules/no-restricted-properties"),
    "no-restricted-syntax": () => require("eslint/lib/rules/no-restricted-syntax"),
    "no-return-assign": () => require("eslint/lib/rules/no-return-assign"),
    "no-return-await": () => require("eslint/lib/rules/no-return-await"),
    "no-script-url": () => require("eslint/lib/rules/no-script-url"),
    "no-self-assign": () => require("eslint/lib/rules/no-self-assign"),
    "no-self-compare": () => require("eslint/lib/rules/no-self-compare"),
    "no-sequences": () => require("eslint/lib/rules/no-sequences"),
    "no-setter-return": () => require("eslint/lib/rules/no-setter-return"),
    "no-shadow": () => require("eslint/lib/rules/no-shadow"),
    "no-shadow-restricted-names": () => require("eslint/lib/rules/no-shadow-restricted-names"),
    "no-spaced-func": () => require("eslint/lib/rules/no-spaced-func"),
    "no-sparse-arrays": () => require("eslint/lib/rules/no-sparse-arrays"),
    "no-sync": () => require("eslint/lib/rules/no-sync"),
    "no-tabs": () => require("eslint/lib/rules/no-tabs"),
    "no-template-curly-in-string": () => require("eslint/lib/rules/no-template-curly-in-string"),
    "no-ternary": () => require("eslint/lib/rules/no-ternary"),
    "no-this-before-super": () => require("eslint/lib/rules/no-this-before-super"),
    "no-throw-literal": () => require("eslint/lib/rules/no-throw-literal"),
    "no-trailing-spaces": () => require("eslint/lib/rules/no-trailing-spaces"),
    "no-undef": () => require("eslint/lib/rules/no-undef"),
    "no-undef-init": () => require("eslint/lib/rules/no-undef-init"),
    "no-undefined": () => require("eslint/lib/rules/no-undefined"),
    "no-underscore-dangle": () => require("eslint/lib/rules/no-underscore-dangle"),
    "no-unexpected-multiline": () => require("eslint/lib/rules/no-unexpected-multiline"),
    "no-unmodified-loop-condition": () => require("eslint/lib/rules/no-unmodified-loop-condition"),
    "no-unneeded-ternary": () => require("eslint/lib/rules/no-unneeded-ternary"),
    "no-unreachable": () => require("eslint/lib/rules/no-unreachable"),
    "no-unsafe-finally": () => require("eslint/lib/rules/no-unsafe-finally"),
    "no-unsafe-negation": () => require("eslint/lib/rules/no-unsafe-negation"),
    "no-unused-expressions": () => require("eslint/lib/rules/no-unused-expressions"),
    "no-unused-labels": () => require("eslint/lib/rules/no-unused-labels"),
    "no-unused-vars": () => require("eslint/lib/rules/no-unused-vars"),
    "no-use-before-define": () => require("eslint/lib/rules/no-use-before-define"),
    "no-useless-call": () => require("eslint/lib/rules/no-useless-call"),
    "no-useless-catch": () => require("eslint/lib/rules/no-useless-catch"),
    "no-useless-computed-key": () => require("eslint/lib/rules/no-useless-computed-key"),
    "no-useless-concat": () => require("eslint/lib/rules/no-useless-concat"),
    "no-useless-constructor": () => require("eslint/lib/rules/no-useless-constructor"),
    "no-useless-escape": () => require("eslint/lib/rules/no-useless-escape"),
    "no-useless-rename": () => require("eslint/lib/rules/no-useless-rename"),
    "no-useless-return": () => require("eslint/lib/rules/no-useless-return"),
    "no-var": () => require("eslint/lib/rules/no-var"),
    "no-void": () => require("eslint/lib/rules/no-void"),
    "no-warning-comments": () => require("eslint/lib/rules/no-warning-comments"),
    "no-whitespace-before-property": () => require("eslint/lib/rules/no-whitespace-before-property"),
    "no-with": () => require("eslint/lib/rules/no-with"),
    "nonblock-statement-body-position": () => require("eslint/lib/rules/nonblock-statement-body-position"),
    "object-curly-newline": () => require("eslint/lib/rules/object-curly-newline"),
    "object-curly-spacing": () => require("eslint/lib/rules/object-curly-spacing"),
    "object-property-newline": () => require("eslint/lib/rules/object-property-newline"),
    "object-shorthand": () => require("eslint/lib/rules/object-shorthand"),
    "one-var": () => require("eslint/lib/rules/one-var"),
    "one-var-declaration-per-line": () => require("eslint/lib/rules/one-var-declaration-per-line"),
    "operator-assignment": () => require("eslint/lib/rules/operator-assignment"),
    "operator-linebreak": () => require("eslint/lib/rules/operator-linebreak"),
    "padded-blocks": () => require("eslint/lib/rules/padded-blocks"),
    "padding-line-between-statements": () => require("eslint/lib/rules/padding-line-between-statements"),
    "prefer-arrow-callback": () => require("eslint/lib/rules/prefer-arrow-callback"),
    "prefer-const": () => require("eslint/lib/rules/prefer-const"),
    "prefer-destructuring": () => require("eslint/lib/rules/prefer-destructuring"),
    "prefer-exponentiation-operator": () => require("eslint/lib/rules/prefer-exponentiation-operator"),
    "prefer-named-capture-group": () => require("eslint/lib/rules/prefer-named-capture-group"),
    "prefer-numeric-literals": () => require("eslint/lib/rules/prefer-numeric-literals"),
    "prefer-object-spread": () => require("eslint/lib/rules/prefer-object-spread"),
    "prefer-promise-reject-errors": () => require("eslint/lib/rules/prefer-promise-reject-errors"),
    "prefer-reflect": () => require("eslint/lib/rules/prefer-reflect"),
    "prefer-regex-literals": () => require("eslint/lib/rules/prefer-regex-literals"),
    "prefer-rest-params": () => require("eslint/lib/rules/prefer-rest-params"),
    "prefer-spread": () => require("eslint/lib/rules/prefer-spread"),
    "prefer-template": () => require("eslint/lib/rules/prefer-template"),
    "quote-props": () => require("eslint/lib/rules/quote-props"),
    quotes: () => require("eslint/lib/rules/quotes"),
    radix: () => require("eslint/lib/rules/radix"),
    "require-atomic-updates": () => require("eslint/lib/rules/require-atomic-updates"),
    "require-await": () => require("eslint/lib/rules/require-await"),
    "require-jsdoc": () => require("eslint/lib/rules/require-jsdoc"),
    "require-unicode-regexp": () => require("eslint/lib/rules/require-unicode-regexp"),
    "require-yield": () => require("eslint/lib/rules/require-yield"),
    "rest-spread-spacing": () => require("eslint/lib/rules/rest-spread-spacing"),
    semi: () => require("eslint/lib/rules/semi"),
    "semi-spacing": () => require("eslint/lib/rules/semi-spacing"),
    "semi-style": () => require("eslint/lib/rules/semi-style"),
    "sort-imports": () => require("eslint/lib/rules/sort-imports"),
    "sort-keys": () => require("eslint/lib/rules/sort-keys"),
    "sort-vars": () => require("eslint/lib/rules/sort-vars"),
    "space-before-blocks": () => require("eslint/lib/rules/space-before-blocks"),
    "space-before-function-paren": () => require("eslint/lib/rules/space-before-function-paren"),
    "space-in-parens": () => require("eslint/lib/rules/space-in-parens"),
    "space-infix-ops": () => require("eslint/lib/rules/space-infix-ops"),
    "space-unary-ops": () => require("eslint/lib/rules/space-unary-ops"),
    "spaced-comment": () => require("eslint/lib/rules/spaced-comment"),
    strict: () => require("eslint/lib/rules/strict"),
    "switch-colon-spacing": () => require("eslint/lib/rules/switch-colon-spacing"),
    "symbol-description": () => require("eslint/lib/rules/symbol-description"),
    "template-curly-spacing": () => require("eslint/lib/rules/template-curly-spacing"),
    "template-tag-spacing": () => require("eslint/lib/rules/template-tag-spacing"),
    "unicode-bom": () => require("eslint/lib/rules/unicode-bom"),
    "use-isnan": () => require("eslint/lib/rules/use-isnan"),
    "valid-jsdoc": () => require("eslint/lib/rules/valid-jsdoc"),
    "valid-typeof": () => require("eslint/lib/rules/valid-typeof"),
    "vars-on-top": () => require("eslint/lib/rules/vars-on-top"),
    "wrap-iife": () => require("eslint/lib/rules/wrap-iife"),
    "wrap-regex": () => require("eslint/lib/rules/wrap-regex"),
    "yield-star-spacing": () => require("eslint/lib/rules/yield-star-spacing"),
    yoda: () => require("eslint/lib/rules/yoda")
}));
